apps/web/src/utils/paymasterSponsor.ts

```diff
@@ -0,0 +1,104 @@
+import { ENTRYPOINT_ADDRESS_V06, UserOperation } from "permissionless";
+import {
+  Address,
+  BlockTag,
+  Hex,
+  decodeAbiParameters,
+  decodeFunctionData,
+} from "viem";
+import { baseSepolia } from "viem/chains";
+import {initializeClient} from "../../src/utils/paymasterConfig";
+
+
+import {
+  coinbaseSmartWalletABI,
+  CB_SW_FACTORY_ADDRESS,
+  CB_SW_PROXY_BYTECODE,
+  CB_SW_V1_IMPLEMENTATION_ADDRESS,
+  ERC_1967_PROXY_IMPLEMENTATION_SLOT,
+  magicSpendAddress
+} from "../constants"
+import RegistrarControllerABI from 'apps/web/src/abis/RegistrarControllerABI';
+import { USERNAME_REGISTRAR_CONTROLLER_ADDRESS } from 'apps/web/src/addresses/usernames';
+
+export async function willSponsor({
+    chainId,
+    entrypoint,
+    userOp,
+}: {chainId: number; entrypoint: string; userOp: UserOperation<"v0.6">}) {
+    if (chainId != baseSepolia.id) return false;
+    if (entrypoint.toLowerCase() !== ENTRYPOINT_ADDRESS_V06.toLowerCase()) return false;
+
+    try {
+        const publicClient = initializeClient();
+        const code = await publicClient.getCode({ address: userOp.sender });
+     
+        if (!code) {
+          // no code at address, check that the initCode is deploying a Coinbase Smart Wallet
+          // factory address is first 20 bytes of initCode after '0x'
+          const factoryAddress = userOp.initCode.slice(0, 42);
```

Comment:
double checking this is doing what you want, since the `0` is going to leave `0x` in the result

Reply:
Yes, the factory address is configured that way as well.